src/components/NcActions/NcActions.vue

```diff
@@ -1008,59 +1019,45 @@ export default {
 		 * @return {Function} the vue render function
 		 */
 		const renderInlineAction = (action) => {
-			const icon = action?.data?.scopedSlots?.icon()?.[0]
-				|| h('span', { class: ['icon', action?.componentOptions?.propsData?.icon] })
-			const attrs = action?.data?.attrs || {}
-			const clickListener = action?.componentOptions?.listeners?.click
-
-			const text = action?.componentOptions?.children?.[0]?.text?.trim?.()
-			const ariaLabel = action?.componentOptions?.propsData?.ariaLabel || text
+			const icon = action?.children?.icon?.()?.[0] || h('span', { class: ['icon', action?.props?.icon] })
+			const clickListener = action?.props?.onClick
+			const text = action?.children?.default?.()?.[0]?.children?.trim()
+			const ariaLabel = action?.props?.['aria-label'] || text
 			const buttonText = this.forceName ? text : ''
 
-			let title = action?.componentOptions?.propsData?.title
+			let title = action?.props?.title
 			// Show a default title for single actions if none is present
 			if (!(this.forceName || title)) {
 				title = text
 			}
 
-			return h('NcButton',
+			const classes = action?.props?.class ? [action?.props?.class, 'action-item action-item--single'] : 'action-item action-item--single'
+			return h(NcButton,
 				{
-					class: [
-						'action-item action-item--single',
-						action?.data?.staticClass,
-						action?.data?.class,
-					],
-					attrs: {
-						...attrs,
-						'aria-label': ariaLabel,
-						title,
-					},
-					ref: action?.data?.ref,
-					props: {
-						// If it has a menuName, we use a secondary button
-						type: this.type || (buttonText ? 'secondary' : 'tertiary'),
-						disabled: this.disabled || action?.componentOptions?.propsData?.disabled,
-						ariaHidden: this.ariaHidden,
-						...action?.componentOptions?.propsData,
-					},
-					on: {
-						focus: this.onFocus,
-						blur: this.onBlur,
-						// If we have a click listener,
-						// we bind it to execute on click and forward the click event
-						...(!!clickListener && {
-							click: (event) => {
-								if (clickListener) {
-									clickListener(event)
-								}
-							},
-						}),
-					},
+					...action?.props,
+					class: classes,
```

Comment:
```js
class: ['action-item', 'action-item--single', action?.props?.class]
```

Reply:
Problem is that `action?.props?.class` can be `undefined` and we would end up with `['action-item', 'action-item--single', undefined]`.

Reply:
Is undefined a problem?

https://github.com/vuejs/core/blob/1789c1caac6fdd3ef3d676d628d7ad6e50888b96/packages/shared/src/normalizeProp.ts#L63-L82

Reply:
Ah, interesting. I thought to have seen an `undefined` in the class list in the DOM, but that's not the case. Let's use it like this then.

Reply:
However, with `mergeProps` its superfluous now :slightly_smiling_face: